### PR TITLE
Ocean Spark clusters

### PR DESCRIPTION
**Pull request by thorsteinnth**
- Adds get, describe, delete operations for Ocean Spark clusters
- Use cluster creation API instead of running a deployer job on the cluster
- Do not attempt to update Ocean controller during cluster creation (error prone and unnecessary)
- bump eksctl-spot dependency
- call eksctl with config file
- cluster + nodegroup created in one go, instead of two separate commands
- remove progress spinner

### CODE REVIEW

File: internal/spot/spotinst_services_ocean_spark.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		TypeMeta: TypeMeta{
+			Kind: typeOf(OceanSparkCluster{}),
+		},
+		ObjectMeta: ObjectMeta{
+			ID:        spotinst.StringValue(output.Cluster.ID),
+			Name:      spotinst.StringValue(output.Cluster.ControllerClusterID),
+			CreatedAt: spotinst.TimeValue(output.Cluster.CreatedAt),
+			UpdatedAt: spotinst.TimeValue(output.Cluster.UpdatedAt),
+		},
+		Obj: output.Cluster,
```
Comment (arnarpall) [inline internal/spot/spotinst_services_ocean_spark.go:78]: Wouldn't we want the `OceanClusterID` and `State` set for single retrievals as well ?

**Review by arnarpall — APPROVED** [review]
Looks great @thorsteinnth

File: internal/spot/spotinst_services_ocean_spark.go (hunk 1 above)
Comment (thorsteinnth) [reply internal/spot/spotinst_services_ocean_spark.go:78]: The single retrieval is only used in the describe operation, which prints out the whole json.
This is consistent with other spotctl get/describe operations.

**Review by arnarpall — APPROVED** [review]

File: internal/dep/defaults.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ var (
 	DependencyEksctlSpot Dependency = &dependency{
 		name:               "eksctl-spot",
 		upstreamBinaryName: "eksctl",
-		version:            "0.77.0-6cfea8af",
+		version:            "0.124.0",
```
Comment (oded7hoffman) [inline internal/dep/defaults.go:27]: @thorsteinnth You can update this to 0.127.0

File: internal/spot/interfaces.go [hunk 3]
```diff
@@ -108,6 +111,21 @@ type (
 		Build() (*OceanRollout, error)
 	}
 
+	// OceanSparkInterface defines the interface of the Spot Ocean for Apache Spark API.
+	OceanSparkInterface interface {
+		// ListClusters returns a list of Ocean Spark clusters.
+		ListClusters(ctx context.Context, controllerClusterID string, state string) ([]*OceanSparkCluster, error)
+
+		// GetCluster returns an Ocean Spark cluster spec by ID.
+		GetCluster(ctx context.Context, clusterID string) (*OceanSparkCluster, error)
```
Comment (oded7hoffman) [inline internal/spot/interfaces.go:120]: @thorsteinnth I'm confused.
ClusterID refers to controllerClusterID or OceanID?
Same for 'DeleteCluster' method

File: internal/cmd/ocean/spark_create_cluster.go [hunk 4]
```diff
@@ -50,6 +46,24 @@ type (
 const (
 	defaultK8sVersion   = "1.21"
```
Comment (oded7hoffman) [inline internal/cmd/ocean/spark_create_cluster.go:47]: @thorsteinnth 
Is it the most updated version we currently support?

**Review by oded7hoffman — APPROVED** [review]

File: internal/cmd/ocean/spark_create_cluster.go (hunk 4 above)
Comment (thorsteinnth) [reply internal/cmd/ocean/spark_create_cluster.go:47]: I will do a separate PR to update the k8s version and eksctl version

File: internal/dep/defaults.go (hunk 2 above)
Comment (thorsteinnth) [reply internal/dep/defaults.go:27]: I will do a separate PR to update the k8s version and eksctl version

File: internal/spot/interfaces.go (hunk 3 above)
Comment (thorsteinnth) [reply internal/spot/interfaces.go:120]: This refers to the Ocean Spark cluster ID (osc-XXX)

**Pull request merged**